### DISCUSSION TITLE
Add instructions to disable admin page or better, make removing ADMIN_TOKEN disable admin page

### DISCUSSION THREAD

**Discussion by ericls** [dani-garcia/vaultwarden#2723] [General]
On the [Enabling admin page](https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page) wiki page, it says adding this `ADMIN_TOKEN` environment variable to enable admin page, and that works. But as a human being, I expect doing the reverse(removing that environment variable) would disable the admin page, but it's not the case. I had to remove the "ADMIN_TOKEN" key in the `config.json` and restart the server to make it work. 

I realize that there's a similar discussion: https://github.com/dani-garcia/vaultwarden/discussions/2164. But that discussion is about how to disable it, while this post is to discuss should we allow disabling it by doing something and suggest a adding a wiki page.

**Comment by ericls** [comment]
To add to the above point, when I googled "vaultwarden disable admin page", the discussion mentioned above didn't pop up.

**Comment by ericls** [comment]
Didn't realize I had edit access to the wiki page, added instructions of disabling admin page to the wiki page.

**Comment by BlackDex** [comment]
Thanks for clarifying on how to disable the admin page after saving the configuration using the admin interface.

**Comment by GlizxyBoi** [comment]
so how do u DO IT!!!!!!!!!!

**Reply by stefan0xC** [reply]
From [the mentioned wiki page](https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page):

> **To disable admin page**, make sure no `ADMIN_TOKEN` environment variable is set, and no `"admin_token"` key exists in `config.json`, if that file exists.

**Reply by Mahh0** [reply]
I had the same issue while updating the ip_header defined as env var. It looks like vaultwarden doesn't read thoses variables if they are already set in config.json, isn't it possible to read it each time the container is started ? Looks like it works with log variables

**Reply by stefan0xC** [reply]
That's because `config.json` takes precedence as explained in the [configuration overview](https://github.com/dani-garcia/vaultwarden/wiki/Configuration-overview#configuration-precedence). Also not all options are saved via the `/admin` panel (i.e. those in the read-only section) so they are unlikely to be overwritten.

**Comment by tantaluzz** [comment]
Just fyi: doing so will crash your Yubikey config - locking your Yubikey users out of their vaults.

So - this is NOT how you do it either.

**Comment by coalfield** [comment]
so how is this done?

**Reply by stefan0xC** [reply]
https://github.com/dani-garcia/vaultwarden/discussions/2723#discussioncomment-4171519

**Reply by stefan0xC** [reply]
Also I've since updated [the wiki page](https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page#disabling-the-admin-page) so it should be even more clear.
